18F/identity-idp
# Show doc auth submission errors at associated field

**Why**: As a user, I expect that if a submission fails, the problem is made obvious by placing an error located at the problematic input field, so that I can most easily work to resolve the issue and resubmit.

**Screenshot:**

![Screen Shot 2020-08-28 at 4 43 09 PM](https://user-images.githubusercontent.com/1779930/91613499-a0ef1c80-e94d-11ea-858b-39fc4557a754.png)

**Testing Instructions:**

Pending testing improvements of LG-3334, this is difficult to test because the current implementation of the React form is eager to disallow non-image files and files smaller than the acceptable minimum.

You must remove this line:

https://github.com/18F/identity-idp/blob/c2170c3a848016af62dbcb67aaba01567a8ea3d9/app/javascript/packages/document-capture/components/acuant-capture.jsx#L187

And start the server as:

```
ACUANT_MINIMUM_FILE_SIZE=0 make run
```

Then upload an error-formatted YML:

```yml
friendly_error: |
  We couldn't read the barcode on the back of your ID. Try taking a new picture.
  Make sure the entire barcode is visible, clean, and doesn't have glares.
  (Error code: 200)
```

And proceed to submit.
